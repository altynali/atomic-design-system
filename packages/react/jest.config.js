/** @type {import('jest').Config} */

const config = {
  verbose: true,
  testEnvironment: "jsdom",
  collectCoverageFrom: ["src/**/*.{js,ts, tsx}"],
  roots: ["<rootDir>/src"],
}

module.exports = config
