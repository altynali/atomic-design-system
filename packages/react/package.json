{
  "name": "atomic-design-system-react",
  "version": "1.0.0",
  "main": "lib/index.js",
  "license": "MIT",
  "devDependencies": {
    "@types/node": "^20.10.0",
    "@types/react": "^18.2.23",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "rollup": "^4.6.0",
    "rollup-plugin-typescript2": "^0.36.0",
    "typescript": "^5.3.2"
  },
  "dependencies": {},
  "scripts": {
    "build": "rollup -c --bundleConfigAsCjs",
    "dev": "yarn build --watch"
  }
}
