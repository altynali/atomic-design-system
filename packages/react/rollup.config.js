import Ts from "rollup-plugin-typescript2"

const config = {
  input: [
    "src/index.ts",
    "src/atoms/Color/index.ts",
    "src/atoms/Text/index.ts",
    "src/atoms/Margin/index.ts",
    "src/atoms/Button/index.ts",
    "src/atoms/Padding/index.ts",
    "src/atoms/Grid/index.ts",
    "src/atoms/Checkbox/index.ts",
    "src/molecules/Card/index.ts",
    "src/molecules/Select/index.ts",
    "src/molecules/Input/index.ts",
    "src/molecules/CardList/index.ts",
    "src/organisms/Form/index.ts",
    "src/templates/Layout/index.ts",
  ],
  output: [
    {
      dir: "lib",
      format: "esm",
      sourcemap: true,
      preserveModules: true,
    },
  ],
  plugins: [Ts()],
  external: ["react", "atomic-design-system-foundation"],
}

export default config
