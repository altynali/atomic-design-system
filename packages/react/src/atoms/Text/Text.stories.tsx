import React from "react"
import Text from "./Text"
import "atomic-design-system-scss/lib/Text.css"

export default {
  title: "Atoms|Text",
}

export const Common = () => <Text>text</Text>
