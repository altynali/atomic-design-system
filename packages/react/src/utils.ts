export const NumberToPixel = (number: number) => {
  return number.toString() + "px"
}

export const widthOrHeightToPixel = (number: number | undefined) => {
  if (typeof number === "number") {
    return NumberToPixel(number)
  }
  return "fit-content"
}

export const classNames = (cls: string, additional: string[] = []): string => {
  const classes = [cls, ...additional?.filter(Boolean)]

  return classes.join(" ")
}
