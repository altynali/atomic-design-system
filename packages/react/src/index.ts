import { Color } from "./atoms/Color"
import { Text } from "./atoms/Text"
import { Margin } from "./atoms/Margin"
import { Button } from "./atoms/Button"
import { Select } from "./molecules/Select"

export { Color, Text, Margin, Button, Select }

// TODO: import * from './atoms';
