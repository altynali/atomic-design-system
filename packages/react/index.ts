import { Color, Margin, Select, Text } from "./src/index"

export { Color, Text, Margin, Select }

// TODO: import * from './atoms';
