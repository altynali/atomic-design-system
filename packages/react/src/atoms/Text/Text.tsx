import React, { FC, PropsWithChildren } from "react"
import { SizesType } from "../../../../foundation/src/types"
import { Sizes } from "atomic-design-system-foundation"

export type TextProps = {
  className?: string
  size?: SizesType
} & PropsWithChildren

const Text: FC<TextProps> = (props) => {
  const { size = Sizes.base, children, className } = props

  const cls = className + ` atds-text atds-text-${size}`

  return <p className={cls}>{children}</p>
}

export default Text
