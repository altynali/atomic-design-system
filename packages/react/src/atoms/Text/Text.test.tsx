import React from "react"
import { render } from "@testing-library/react"
import Text from "./Text"
import { Sizes } from "atomic-design-system-foundation/lib/src"
import "@testing-library/jest-dom"

describe("Text component", () => {
  it("renders with default size", () => {
    const { getByText } = render(<Text>Hello World</Text>)
    const textElement = getByText("Hello World")
    expect(textElement).toBeInTheDocument()
    expect(textElement).toHaveClass("atds-text")
    expect(textElement).toHaveClass("atds-text-base")
  })

  it("renders with custom size", () => {
    const customSize = Sizes.sm
    const { getByText } = render(<Text size={customSize}>Hello World</Text>)
    const textElement = getByText("Hello World")
    expect(textElement).toBeInTheDocument()
    expect(textElement).toHaveClass("atds-text")
    expect(textElement).toHaveClass(`atds-text-${customSize}`)
  })

  test("snapshot of component", () => {
    const { asFragment } = render(<Text>Hello World</Text>)

    expect(asFragment()).toMatchSnapshot()
  })
})
