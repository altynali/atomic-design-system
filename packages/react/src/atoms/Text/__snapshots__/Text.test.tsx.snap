// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component snapshot of component 1`] = `
<DocumentFragment>
  <p
    class="undefined atds-text atds-text-base"
  >
    Hello World
  </p>
</DocumentFragment>
`;
